src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPiRegisterDriver.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Device.Gpio.Drivers;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Device.Gpio
+{
+    internal class RaspberryPiRegisterDriver : GpioDriver
+    {
+        private GpioDriver _internalDriver;
+
+        public RaspberryPiRegisterDriver()
+        {
+            _internalDriver = new Windows10Driver();
+        }
+
+        protected internal virtual ulong ClearRegister
+        {
+            get { throw new PlatformNotSupportedException(); }
+            set { throw new PlatformNotSupportedException(); }
+        }
+
+        protected internal virtual ulong SetRegister
+        {
+            get { throw new PlatformNotSupportedException(); }
+            set { throw new PlatformNotSupportedException(); }
+        }
```

Comment:
Do we need these to be declared here? They're only used in the derived class.

Reply:
They are also used in Windows when using the raspberrypi3driver. We could remove them if we decided to add an ifcheck on RaspberryPi3Driver everytime these methods get called but we don’t want to add another perf hit on this critical scenario. One thing I didn’t explore was to just have these two be delegate functions that get set on the constructor of raspberrypi3driver, that way we only have to do the check once. I’ll try exploring that option as well.

Reply:
Actually, I wonder what these are intended for. Our code nowhere uses these properties and they would only allow a derived class to access a very specific register.

Reply:
These two properties are basically a shortcut we had to create in order to support the RGBMatrix device binding. We only use them there: https://github.com/dotnet/iot/blob/master/src/devices/RGBLedMatrix/Gpio.cs

Basically this device requires you to be able to write to GPIO at a super high rate, and by adding these properties we basically allow you to write a) more than one pin at a time and b) skip validation overhead we do when you try to write to a pin using the controller. You can only call these two if you manually passed in the driver when creating the GpioController, and you keep a handle to the driver since you call them via the driver. They are also not super easy to use, since they are modifying the registers inside the Pi directly in order to set the value of the pins.

Reply:
I see. Maybe providing the pointer to /dev/gpiomem would be more flexible and even easier to use (for bindings that know what they're doing). And redirecting would not have a performance impact.

Reply:
That is more or less what this is doing, except that instead of giving back the whole gpiomem, it only gives back the specific registers for changing the gpio values.